tests/integration/conftest.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+
+import pytest_asyncio
+from pytest_operator.plugin import OpsTest
+
+
+@pytest_asyncio.fixture(scope="module", name="get_unit_ips")
+async def fixture_get_unit_ips(ops_test: OpsTest):
+    """Return an async function to retrieve unit ip addresses of a certain application."""
+
+    async def get_unit_ips(application_name: str):
+        """Retrieve unit ip addresses of a certain application.
+
+        Returns:
+            a list containing unit ip addresses.
+        """
+        _, status, _ = await ops_test.juju("status", "--format", "json")
+        status = json.loads(status)
+        units = status["applications"][application_name]["units"]
+        return tuple(
+            unit_status["address"]
+            for _, unit_status in sorted(units.items(), key=lambda kv: int(kv[0].split("/")[-1]))
```

Comment:
Out of curiosity, may I ask why the sort?

Reply:
This is to ensure that the return list is consistent, probably not very useful in here.